python/nav/web/maintenance/views.py

```diff
@@ -54,14 +54,16 @@ def redirect_to_calendar(_request):
 
 def calendar(request, year=None, month=None):
     # If the form was used to get here, redirect to the appropriate page
+    form = MaintenanceCalendarForm()
+    heading = "Maintenance schedule"
+
     if "year" in request.GET and "month" in request.GET:
         return redirect(
             "maintenance-calendar",
             year=request.GET.get("year"),
             month=request.GET.get("month"),
         )
```

Comment:
This code path is still not actually validating the input, meaning this doesn't actually fix #2420. You need to feed the input into a `MaintenanceCalendarForm` to validate. 

If the form validates, you can redirect the user, being safe that their input should match the `maintenance-calendar` URL pattern. This is the "happy" path.

I.e. something like 
```python
form = MaintenanceCalendarForm(data=request.GET.dict())
if form.is_valid():
  # happy happy joy joy, redirect!
```

If the form doesn't validate, you should move on and render the page as usual, using the form you initialized with the data.

If `year` and `month` weren't present as GET arguments, the form instance should be instead created as e.g.:

```python
form = MaintenanceCalendarForm(
    data={'year': year, 'month': month}, 
    initial={'year': this_year, 'month': this_month}
)
```